test/torchaudio_unittest/prototype/hifi_gan/hifi_gan_test_impl.py

```diff
@@ -134,3 +107,26 @@ def test_original_implementation_match(self):
         ref_output = model_ref(inputs)
         output = model(inputs)
         self.assertEqual(ref_output, output)
+
+    def test_mel_transform(self):
+        """Check that HiFiGANMelSpectrogram generates same mel spectrogram as the original HiFiGAN implementation
+        when applied on a synthetic waveform.
+        There seems to be no way to change dtype in the original implmentation, so we feed in the waveform with the
```

Comment:
I feel it's okay to change dtype in `ref_mel_spectrogram`, which part cannot change the dtype of tensor?

Reply:
If I feed `float64` into `ref_mel_spectrogram`, it throws an error 
```
RuntimeError: expected scalar type Double but found Float
```
on the line `spec = torch.matmul(mel_basis[str(fmax)+'_'+str(y.device)], spec)`. I think this is because `mel_basis` is created with explicit type `float` [here](https://github.com/jik876/hifi-gan/blob/4769534d45265d52a904b850da5a622601885777/meldataset.py#L58). Now that we've copied the code into our repo we can of course modify it and enable passing `float64`. Do we want to do it or do we want to keep the copied code as close as possible to the original?